core/trino-spi/src/main/java/io/trino/spi/block/RowBlockBuilder.java

```diff
@@ -154,7 +156,18 @@ public SingleRowBlockWriter beginBlockEntry()
             throw new IllegalStateException("Expected current entry to be closed but was opened");
         }
         currentEntryOpened = true;
-        return new SingleRowBlockWriter(fieldBlockBuilders[0].getPositionCount(), fieldBlockBuilders);
+        singleRowBlockWriter.reset(fieldBlockBuilders[0].getPositionCount());
+        return singleRowBlockWriter;
+    }
+
+    // It is only draft to be throw away. It is the fastest way to compare benchmarks. If results will be promising, it will be written prettier.
+    public SingleRowBlockWriter beginBlockEntryAlloc()
+    {
+        if (currentEntryOpened) {
+            throw new IllegalStateException("Expected current entry to be closed but was opened");
+        }
+        currentEntryOpened = true;
+        return new SingleRowBlockWriter(fieldBlockBuilders[0].getPositionCount(), this.fieldBlockBuilders);
     }
```

Comment:
`closeEntry` should set `singleRowBlockWriter` position to invalid one (e.g. -1) so that `singleRowBlockWriter` cannot be used without `beginBlockEntry`